src/arti/tor-dirmgr/src/state.rs

```diff
@@ -96,18 +94,23 @@ pub(crate) struct GetConsensusState<DM: WriteNetDir> {
 impl<DM: WriteNetDir> GetConsensusState<DM> {
     /// Create a new GetConsensusState from a weak reference to a
     /// directory manager and a `cache_usage` flag.
-    pub(crate) fn new(writedir: Weak<DM>, cache_usage: CacheUsage) -> Self {
-        let authority_ids: Vec<_> = default_authorities()
-            .iter()
-            .map(|auth| *auth.v3ident())
-            .collect();
-
-        GetConsensusState {
+    pub(crate) fn new(writedir: Weak<DM>, cache_usage: CacheUsage) -> Result<Self> {
+        let authority_ids: Vec<_> = if let Some(writedir) = Weak::upgrade(&writedir) {
+            writedir
+                .config()
+                .authorities()
+                .iter()
+                .map(|auth| *auth.v3ident())
+                .collect()
+        } else {
+            return Err(Error::ManagerDropped.into());
+        };
```

Comment:
You can avoid some indent by going even more functional :)

```suggestion
        let authority_ids: Vec<_> = Weak::upgrade(&writedir)
            .context(Error::ManagerDropped)?
            .config()
            .authorities()
            .iter()
            .map(|auth| *auth.v3ident())
            .collect();
```

Reply:
Whooo! That's great!